internal/versions/versions.go

```diff
@@ -115,21 +115,27 @@ func (h *handler) GetReleaseImage(ctx context.Context, openshiftVersion, cpuArch
 	if err := h.kubeClient.List(ctx, clusterImageSets); err != nil {
 		return nil, err
 	}
+	var wg sync.WaitGroup
 	for _, clusterImageSet := range clusterImageSets.Items {
-		existsInCache := false
-		for _, releaseImage := range h.releaseImages {
-			if releaseImage.URL != nil && *releaseImage.URL == clusterImageSet.Spec.ReleaseImage {
-				existsInCache = true
-				break
+		wg.Add(1)
+		go func(clusterImageSet hivev1.ClusterImageSet) {
```

Comment:
Right now this is spawning all the routines simultaneously, but I can also do this in batches if we think this is too much.

As is this would pull and inspect all 80 release images in parallel. While it works, I'm worried that it might not scale very well.

What do others think?

Reply:
+1 for batches, however we don't support a lot of old OpenShift releases, so its unlikely that current 80 release images will grow significantly. I wonder if we can test it on 200 release images to be sure